owncloudApp/src/main/java/com/owncloud/android/ui/activity/FileDisplayActivity.kt

```diff
@@ -1151,20 +1124,26 @@ class FileDisplayActivity : FileActivity(), FileFragment.ContainerActivity, OnEn
         setFile(file)
     }
 
-    override fun updateActionBarTitleAndHomeButton(chosenFileFromParam: OCFile?) {
+    fun updateActionBarTitleAndHomeButton(chosenFileFromParam: OCFile?) {
         var chosenFile = chosenFileFromParam
         if (chosenFile == null) {
             chosenFile = file     // if no file is passed, current file decides
         }
-        super.updateActionBarTitleAndHomeButton(chosenFile!!)
-        if (chosenFile?.remotePath == OCFile.ROOT_PATH && (!fileListOption.isAllFiles())) {
+        if (chosenFile?.remotePath == OCFile.ROOT_PATH) {
             val title =
-                when {
-                    fileListOption.isAvailableOffline() -> resources.getString(R.string.drawer_item_only_available_offline)
-                    fileListOption.isSharedByLink() -> resources.getString(R.string.drawer_item_shared_by_link_files)
-                    else -> null // Should not happen -> will show R.string.app_name
+                when (fileListOption) {
+                    FileListOption.AV_OFFLINE -> resources.getString(R.string.drawer_item_only_available_offline)
+                    FileListOption.SHARED_BY_LINK -> resources.getString(R.string.drawer_item_shared_by_link_files)
+                    FileListOption.ALL_FILES -> resources.getString(R.string.default_display_name_for_root_folder)
                 }
-            updateActionBarTitleAndHomeButtonByString(title)
+            setupToolbar(ToolbarConfig.ToolbarRoot(title, true))
+            listOfFilesFragment?.setSearchListener(findViewById(R.id.root_toolbar_search_view))
+        } else {
+            setupToolbar(ToolbarConfig.ToolbarStandard(
+                title = chosenFile?.fileName!!,
```

Comment:
!! 😮

Reply:
Fixed in a nice way from my point of view.